### PR TITLE
Add method of testing gameplay from editor

### PR DESCRIPTION
**Pull request by bdach**
This PR contains the *minimum viable prototype* of #7008. Many things and quality of life features are missing. Those will come later if the initial MVP goes through.

I am primarily PRing for comment and expect there to be iterations on this pull.

https://user-images.githubusercontent.com/20418176/141358091-5da37af1-e365-4872-a900-3285a855ef51.mp4

Key pre-emptions:

* As it turns out, with the save-before-preview cheat, `Player` mostly works with this. One thing that I've had to locally hack around is the fact that the `ShowResults` flag [will ensure results are not reached, but will not exit out of the player either](https://github.com/ppy/osu/blob/3b7a8f9d580ed0254913289cf642135b3bb5bd9c/osu.Game/Screens/Play/Player.cs#L685-L686). So I've had to manually exit on score processor completion. This isn't 100% correct with regard to storyboard outros and is intended to be "good enough for now". I am willing to listen to ideas on how to reshape `Player` to accomodate this.

  Ideally I'd have [`progressToResults()`](https://github.com/ppy/osu/blob/3b7a8f9d580ed0254913289cf642135b3bb5bd9c/osu.Game/Screens/Play/Player.cs#L732-L741) as virtual and exit player there. But having both that and `ShowResults` seems wrong, so I'd probably have to replace existing usages of `ShowResults` with overriding that method which may end up being a bit overlong.
* The design for the button is a combination of the current editor's visual language [and the new design](https://www.figma.com/file/ytnnne2TH8Z956Jxhiypqq/Beatmap-Editor-2?node-id=0%3A1). The colours and typography from the new design are there, the things that don't match are the corner radius and the paddings. I can handle that as a follow-up to this, by restyling the entire bottom bar to match design.

  The move of `OverlayColourProvider` to `Editor` level is supposed to accomodate future redesigns.
* Again, I am aware of several missing functionalities (sharing clock state with the player, e.g. being able to start gameplay from any point, or things like being able to toggle autoplay). I will look into all of that once agreement on this MVP is reached.

### CODE REVIEW

**Comment by peppy** [comment]
Pretty good start to this! If I had to pick on anything, it would be the `PlayerLoader` taking longer than I'd want it to when testing a beatmap. Maybe it's fine to leave it as is until we integrate the player transition better into the editor? If people complain about it would could make an `EditorPlayerLoader` with shorter transitions (or skipping some completely), waiting the bare minimum time required to perform the beatmap reload.

Happy to see this go as-is though.

**Review by peppy — DISMISSED** [review]

**Comment by bdach** [comment]
Noted, I'll look into reducing the loader transition time.

**Comment by peppy** [comment]
If you do that, I'd propose the osu! logo also doesn't do any flying animations and is already positioned in the centre (right now it can either fly from the song select bottom right on first test, or move slightly vertically on subsequent tests).

I'd be particular wanting to see the initial animation skipped where the metadata is revealed. It should already be revealed at the point of the whole loader screen fading in.

File: osu.Game/Screens/Edit/Editor.cs [hunk 1]
```diff
@@ -456,6 +468,10 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                     menuBar.Mode.Value = EditorScreenMode.Verify;
                     return true;
 
+                case GlobalAction.EditorTestGameplay:
+                    testGameplay();
```
Comment (Joehuu) [inline osu.Game/Screens/Edit/Editor.cs:472]: We usually click the button instead for feedback (once https://github.com/ppy/osu/issues/14132 is addressed though).

Comment (peppy) [reply osu.Game/Screens/Edit/Editor.cs:472]: Seems valid, will change that here.

**Review by peppy — APPROVED** [review]

**Pull request merged**